pkg/performanceprofile/controller/performanceprofile/hypershift/hypershift.go

```diff
@@ -65,36 +75,59 @@ func NewControlPlaneClient(c client.Client, ns string) *ControlPlaneClientImpl {
 	}
 }
 
-func (ci *ControlPlaneClientImpl) getFromConfigMap(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
-	cm := &corev1.ConfigMap{}
+func (ci *ControlPlaneClientImpl) getFromConfigMap(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) (bool, error) {
+	var labelSelector labels.Selector
+	switch obj.(type) {
+	case *performancev2.PerformanceProfile:
+		labelSelector = labels.SelectorFromSet(labels.Set{hypershiftControllerGeneratedPerformanceProfile: "true"})
+	case *machineconfigv1.KubeletConfig:
+		labelSelector = labels.SelectorFromSet(labels.Set{ConfigMapEncapsulatedKindKey: "kubeletconfig"})
+	case *machineconfigv1.MachineConfig:
+		labelSelector = labels.SelectorFromSet(labels.Set{ConfigMapEncapsulatedKindKey: "machineconfig"})
+	case *tunedv1.Tuned:
+		labelSelector = labels.SelectorFromSet(labels.Set{ConfigMapEncapsulatedKindKey: "tuned"})
+	default:
+		return false, nil
+	}
+	cmList := &corev1.ConfigMapList{}
+	listOpts := &client.ListOptions{}
 	if key.Namespace == metav1.NamespaceNone {
-		key.Namespace = ci.managementClusterNamespaceName
+		listOpts.Namespace = ci.managementClusterNamespaceName
+		listOpts.LabelSelector = labelSelector
 	}
-	err := ci.Client.Get(ctx, key, cm, opts...)
+	err := ci.List(ctx, cmList, listOpts)
```

Comment:
is the object name subject to change or once we discovered initially the right object name we can safely assume it's going to be stable?
I guess it boils down to the question if we can cache the information we discover here.

Reply:
The name in this context is the name of the encapsulated object, so changing the name of the object means, changing the `data` in the ConfigMap, which is something that we can cache and track for changes.

Reply:
ok. I'm trying to explore this direction: there could be a predictable and fixed relationship between the outer ConfigMap name and the inner object name/kind? if not, do we know why?

Reply:
That's depends - 
There are ConfigMaps which are controller fully by NTO controls and there other which are not.
All the object listed below are of course encapsulated in ConfigMaps:

A. Fully controlled by NTO - 
     1. KubletConfig 
     2. MachineConfig
     3. Tuned

B. Not controlled by NTO - 
    1. PerformanceProfile
    2. ContainerRuntimeConfig
    3. Tuned (which applied directly by the user)

ConfigMap that are fully controlled by NTO (group A) has a predicted names with fixed relationship between the internal object and the ConfigMap.

The others (group B) are being provided by the user and mirrored into the hosted control plane namespace (the namespace where NTO operator's pod resides) and we cannot control the naming convention of those.

Reply:
> B. Not controlled by NTO - 1. PerformanceProfile 2. ContainerRuntimeConfig 3. Tuned (which applied directly by the user)
> 
> ConfigMap that are fully controlled by NTO (group A) has a predicted names with fixed relationship between the internal object and the ConfigMap.
> 
> The others (group B) are being provided by the user and mirrored into the hosted control plane namespace (the namespace where NTO operator's pod resides) and we cannot control the naming convention of those.

OK, so I'm rusty on hypershift but thing is: there is still some option because the mirroring agent (hypershift) does control the object creation into the hosted control plane.
However thinking twice this is not helping much because in order to preserve the original metadata (e.g. name) that controller would likely need to use labels/annotations, which is pretty much the same thing we're doing already :\